core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -181,9 +186,7 @@ private[spark] class TaskSetManager(
   private def addPendingTask(index: Int) {
     // Utility method that adds `index` to a list only if it's not already there
     def addTo(list: ArrayBuffer[Int]) {
-      if (!list.contains(index)) {
-        list += index
-      }
+      list += index
     }
```

Comment:
You're in luck because I accidentally merged before Jenkins got to it, so we reverted it pending tests. There's a new PR here: https://github.com/apache/spark/pull/11175

Would you mind sticking this comment there? Agree with your sentiment.